cscs-checks/libraries/io/netcdf_compile_run.py

```diff
@@ -9,26 +9,37 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(*([lang, linkage] for lang in ['cpp', 'c', 'f90']
-                          for linkage in ['dynamic', 'static']))
+@rfm.simple_test
 class NetCDFTest(rfm.RegressionTest):
-    def __init__(self, lang, linkage):
-        lang_names = {
+    lang = parameter(['cpp', 'c', 'f90'])
+    linkage = parameter(['dynamic', 'static'])
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                     'arolla:cn', 'tsa:cn']
+    build_system = 'SingleSource'
+    num_tasks = 1
+    num_tasks_per_node = 1
+    maintainers = ['AJ', 'SO']
+    tags = {'production', 'craype', 'external-resources', 'health'}
+    lang_names = {
             'c': 'C',
             'cpp': 'C++',
             'f90': 'Fortran 90'
-        }
-        self.lang = lang
-        self.linkage = linkage
-        self.descr = f'{lang_names[lang]} NetCDF {linkage.capitalize()}'
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'arolla:cn', 'tsa:cn']
-        if linkage == 'dynamic':
+    }
+
+    @run_after('init')
+    def set_description(self):
+        self.descr = (f'{self.lang_names[self.lang]} NetCDF '
+                      f'{self.linkage.capitalize()}')
+
+    @run_after('init')
+    def setup_prgenvs(self):
+        if self.linkage == 'dynamic':
             self.valid_systems += ['eiger:mc', 'pilatus:mc']
 
         if self.current_system.name in ['daint', 'dom']:
             self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
-                                        'PrgEnv-intel', 'PrgEnv-pgi']
+                                        'PrgEnv-intel', 'PrgEnv-pgi',
+                                        'PrgEnv-nvidia']
```

Comment:
Same as hdf5: add `osext.cray_cdt_version() < '21.05',` for PrgEnv-nvidia/PrgEnv-pgi